### PR TITLE
Pivotal ID # 176886357: Async Submission Request

### PR DESCRIPTION
**Pull request by jhoanmanuelms**
https://www.pivotaltracker.com/story/show/176886357

- Store files full path in Mongo documents
- Adjust the files service in order to process  submission files recursively  and update them with the proper path

### CODE REVIEW

File: submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoPersistenceService.kt [hunk 1]
```diff
@@ -49,17 +47,18 @@ internal class SubmissionMongoPersistenceService(
 
     override fun processSubmission(saveRequest: SaveSubmissionRequest): ExtSubmission {
         val (submission, fileMode, accNo) = saveRequest
-        subDataRepository.updateStatus(PROCESSING, accNo, submission.version)
-        systemService.persistSubmissionFiles(submission, fileMode)
-        processDbSubmission(submission)
+//        subDataRepository.updateStatus(PROCESSING, accNo, submission.version)
```
Comment (Juan-EBI) [inline submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoPersistenceService.kt:50]: ?

File: submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoPersistenceService.kt [hunk 2]
```diff
@@ -31,8 +29,8 @@ internal class SubmissionMongoPersistenceService(
         val submission = saveRequest.submission
         val newVersion = submission.copy(
             version = getNextVersion(submission.accNo),
-            status = ExtProcessingStatus.REQUESTED)
-        subDataRepository.save(newVersion.toDocSubmission())
+            status = REQUESTED)
+//        subDataRepository.save(newVersion.toDocSubmission())
```
Comment (Juan-EBI) [inline submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoPersistenceService.kt:33]: ?

File: submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoPersistenceService.kt (hunk 1 above)
Comment (jhoanmanuelms) [reply submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoPersistenceService.kt:50]: Nope 😅

File: submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoPersistenceService.kt (hunk 2 above)
Comment (jhoanmanuelms) [reply submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoPersistenceService.kt]: fixed

File: submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoPersistenceService.kt (hunk 1 above)
Comment (jhoanmanuelms) [reply submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoPersistenceService.kt]: fixed

**Review by Juan-EBI — APPROVED** [review]

**Pull request merged**